packages/js/admin-layout/src/gutenberg/index.ts

```diff
@@ -0,0 +1,12 @@
+export * from './admin-layout';
+export * from './components';
+
+/**
+ * External dependencies
+ */
+/* eslint-disable @woocommerce/dependency-group */
+// @ts-ignore No types for this exist yet.
+import { unlock } from '@wordpress/edit-site/build-module/lock-unlock';
+/* eslint-enable @woocommerce/dependency-group */
+
+export { unlock };
```

Comment:
Instead of exporting `unlock`, what if the `admin-layout` package exported the unlocked comonents instead? This `unlock` function is used solely to access the `@wordpress/router` package and its `RouterProvider`, `useHistory` and `useLocation` exports. Let's re-export them directly. Then all the problematic private API accesses are concentrated at one place.

An alternative to having a `@wordpress/router` package with private exports would be _bundling_ this package. There doesn't really need to be a `wp-router` script and `wp.router` global, at least not yet. The router code could be bundled with the app that uses it.

In Gutenberg the `router` package is used exclusively by the `edit-site` app, and could also be bundled together with it. With one exception: there are commands in `core-commands` that navigate to a Site Editor page using `history.navigate`. This could be replaces with a low-level `history.pushState` call. The commands don't do any transitions etc.

What do you think @youknowriad? We'll only need a global `wp.router` when multiple admin pages are merged into one SPA and do navigations between each other without page reloads. And at that time it should be stable and public.